### PR TITLE
Fix scratch n push

### PR DESCRIPTION
**Pull request by hatchan**
Also fixes some issues with `--docker-local`

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -1019,39 +1022,37 @@ func (s *DockerPushStep) Execute(ctx context.Context, sess *core.Session) (int,
 }
 
 func (s *DockerPushStep) tagAndPush(imageID string, e *core.NormalizedEmitter, client *DockerClient, auth docker.AuthConfiguration) (int, error) {
-	if !s.dockerOptions.DockerLocal {
```
Comment (hatchan) [inline docker/docker.go]: Just removed the check, which caused everything else to collapse (there shouldn't be any changes in the code below)

**Comment by Tonnu** [comment]
lgtm :+1:

**Comment by hatchan** [comment]
Tested it locally, and it worked

**Comment by flenter** [comment]
lgtm

**Pull request merged**